### PR TITLE
Add 3D fill, "mill", and "print" on top of #3074

### PR DESCRIPTION
**Pull request by jni**
# Description

This adds our first 3D "write" interactivity! 🎉 Currently this PR is on top of #3074, it should be rebased after that merges, but I couldn't wait to play with it. Short video below.

Not everything is perfect yet but it's a brilliant start. The biggest issue I've noticed playing with it is that interpolating in 3D while filling is dangerous. 😂 Perhaps filling label 0 should be disabled in 3D?

Here's a video of it in action:

https://user-images.githubusercontent.com/492549/127662770-d765c8b7-92b2-4b8f-971b-73df67ab0753.mov

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by jni** [comment]
You can look at 9c21ce2 for the changes from this PR... It's pretty minimal! 🎉

**Review by alisterburt — COMMENTED** [review]
Wow! I LOVE how simple the implementation is, really feels like we're onto a winner with the exposed interface. Amazing stuff 😍 can't wait to see this merged. 

At first pass I see no reason not to merge this as soon as #3074 is in? Ping me for review when you think this is ready!

We should make sure to use your `first_nonzero_coordinate()` utility in the labels layer `Labels.get_value()` method in #3074 too!

**Comment by jni** [comment]
@alisterburt feel free to "approve" the PR. ;) It's ready now as far as I'm concerned!

**Review by alisterburt — APPROVED** [review]
heartily approved!

**Review by kevinyamauchi — COMMENTED** [review]
Thanks @jni ! This is super neat. The code looks good to me. I think it would be nice to at least have tests for the new `_labels_utils` functions and maybe for the Qt mode switching. If you don't have time, I can probably work on it some this week.

File: napari/layers/labels/_labels_utils.py [hunk 1]
```diff
@@ -133,3 +137,68 @@ def get_dtype(layer):
...
+    coords = np.linspace(start_point, end_point, length_int + 1, endpoint=True)
+    clipped_coords = np.clip(coords, 0, shape - 1).astype(int)
+    nonzero = np.flatnonzero(data[tuple(clipped_coords.T)])
+    if len(nonzero) == 0:
+        return None
+    else:
+        return clipped_coords[nonzero[0]]
+
+
+def get_action_coordinate(layer, event):
```
Comment (sofroniewn) [inline napari/layers/labels/_labels_utils.py]: Can this maybe be `get_data_coordinate`? I find `action` quite confusing as this isn't an action it's getting - it's a vispy event

**Review by sofroniewn — APPROVED** [review]
I left a request for a name change, also agree with reviewer above, tests would be nice!

**Comment by jni** [comment]
Ok @sofroniewn @kevinyamauchi @alisterburt tests are added! 😅 This should be ready. 🤞

**Review by sofroniewn — APPROVED** [review]
LGTM! Merge when ready!!

**Comment by jni** [comment]
(the failing run is unrelated to this — all tests are passing even on minimum requirements.)

**Pull request merged**